apm-agent-plugins/apm-okhttp-plugin/src/main/java/co/elastic/apm/agent/okhttp/OkHttp3ClientInstrumentation.java

```diff
@@ -40,62 +36,52 @@
 import okhttp3.HttpUrl;
 import okhttp3.Request;
 
+import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 
 import static net.bytebuddy.matcher.ElementMatchers.named;
 import static net.bytebuddy.matcher.ElementMatchers.returns;
 
 public class OkHttp3ClientInstrumentation extends AbstractOkHttp3ClientInstrumentation {
 
-    public OkHttp3ClientInstrumentation(ElasticApmTracer tracer) {
-        super(tracer);
-    }
-
     @Override
     public Class<?> getAdviceClass() {
         return OkHttpClient3ExecuteAdvice.class;
     }
 
-    @VisibleForAdvice
     public static class OkHttpClient3ExecuteAdvice {
 
-        @VisibleForAdvice
-        public final static GlobalThreadLocal<Span> spanTls = GlobalThreadLocal.get(OkHttpClient3ExecuteAdvice.class, "spanTls");
```

Comment:
It's probably fine either way, so just out of curiosity, why did you change from `GlobalThreadLocal<Span>` to returning the span in the `Object[]` return value?

Reply:
Hah, good question :) I don't know why.
What is the difference between using GlobalThreadLocal and returning an array of objects? And when to use them?

Reply:
Also a good question :)
I guess in terms of allocations, the `Obect[]` approach might even be better, as the JIT compiler should be able to do some escape analysis and replace it with a stack allocation. Also, it seems a bit safer maybe as there are fewer things that can go wrong, such as forgetting to clear the ThreadLocal.
A downside is that it requires more casting and things can go wrong when accessing the wrong index in the array.

But overall, the `Object[]` approach is probably preferable in most situations.